feat: Add Moorcheh vector store integration

bhavanipot opened the pull request:
# Description

This pull request adds an official integration for Moorcheh, a semantic vector database developed by EdgeAI Innovations, under the llama-index-integrations/vector_stores module. The integration adheres to the standard vector store interface defined in LlamaIndex and provides support for core methods such as add, query, delete, and search.

The implementation follows the design patterns established by existing vector store integrations in the repository. The integration is self-contained within llama-index-integrations/vector_stores/llama-index-vector-stores-moorcheh.

We look forward to feedback from the maintainers and are happy to make any required adjustments.

Fixes # (issue): N/A

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [x] I added new unit tests to cover this change
- [ ]  I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `uv run make format; uv run make lint` to appease the lint gods

AstraBert reviewed (commented):
PR is very cool, but needs some adjustments (see comments 👇). Good work tho!

bhavanipot commented:
We’ve addressed all the feedback from the initial review, including updating the implementation and fixing the failing tests. The integration now supports the requested changes - our code is now ready for re-review. Thanks again!

AstraBert commented on llama-index-integrations/vector_stores/llama-index-vector-stores-moorcheh/llama_index/vector_stores/moorcheh/base.py:
```diff
@@ -0,0 +1,461 @@
+        return "MoorchehVectorStore"
+
+    @property
+    def client(self) -> MoorchehClient:
```
Duplicate ;)

AstraBert commented on llama-index-integrations/vector_stores/llama-index-vector-stores-moorcheh/tests/more_tests.py:
Please rename this file so that it starts with `test_`, thanks :)

AstraBert reviewed (changes requested):
Please change the name of the test file :)

When you do, ping me (just mention me) and I'll merge the PR

AstraBert commented:
@bhavanipot 
@ShatterpointStar 

Please mention me when you change the name of the test file, so that I can merge the PR

bhavanipot commented:
@AstraBert Thank you for the review! We've renamed the file as requested — please let me know if there's anything else you'd like us to adjust.

AstraBert commented:
No need for any more changes, once the CI runs through I'll merge it :))

AstraBert commented:
Getting this errorr in tests:

```text
==================================== ERRORS ====================================
____________ ERROR collecting tests/test_vector_store_moorcheh_2.py ____________
tests/test_vector_store_moorcheh_2.py:20: in <module>
    @pytest.mark.skipif(should_skip, reason="MOORCHEH_API_KEY not set")
E   NameError: name 'should_skip' is not defined
=========================== short test summary info ============================
ERROR tests/test_vector_store_moorcheh_2.py - NameError: name 'should_skip' is not defined
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
1 error in 2.69s
```

ShatterpointStar commented:
@AstraBert Hi, we have fixed the error so it should be ready now, thank you.
